core/src/main/java/org/jruby/truffle/nodes/rubinius/TimePrimitiveNodes.java

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (c) 2015 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.jruby.truffle.nodes.rubinius;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.source.SourceSection;
+import org.joda.time.DateTime;
+import org.jruby.truffle.nodes.objectstorage.ReadHeadObjectFieldNode;
+import org.jruby.truffle.nodes.objectstorage.WriteHeadObjectFieldNode;
+import org.jruby.truffle.runtime.RubyContext;
+import org.jruby.truffle.runtime.core.*;
+import org.jruby.util.ByteList;
+import org.jruby.util.RubyDateFormatter;
+
+/**
+ * Rubinius primitives associated with the Ruby {@code Time} class.
+ * <p>
+ * Also see {@link RubyTime}.
+ */
+public abstract class TimePrimitiveNodes {
+
+    @RubiniusPrimitive(name = "time_s_now")
+    public static abstract class TimeSNowPrimitiveNode extends RubiniusPrimitiveNode {
+
+        public TimeSNowPrimitiveNode(RubyContext context, SourceSection sourceSection) {
+            super(context, sourceSection);
+        }
+
+        public TimeSNowPrimitiveNode(TimeSNowPrimitiveNode prev) {
+            super(prev);
+        }
+
+        @Specialization
+        public RubyTime timeSNow(RubyClass timeClass) {
+            final long milliseconds = System.currentTimeMillis();
+            return new RubyTime(timeClass,
+                            TimeOperations.millisecondsToSeconds(milliseconds),
+                            TimeOperations.millisecondsToNanoseconds(TimeOperations.millisecondsInCurrentSecond(milliseconds)));
+        }
+
+    }
+
+    @RubiniusPrimitive(name = "time_s_dup", needsSelf = false)
+    public static abstract class TimeSDupPrimitiveNode extends RubiniusPrimitiveNode {
```

Comment:
It sets the native fields of seconds and nanoseconds for one. Then it just seemed easier to directly copy the fields, but now I think about a simple call to `clone` might be better.